mapsforge-map-reader/src/main/java/org/mapsforge/map/reader/MapFile.java

```diff
@@ -184,6 +188,9 @@ public class MapFile extends MapDataStore {
     private byte zoomLevelMin = 0;
     private byte zoomLevelMax = Byte.MAX_VALUE;
 
+    // container for loaded tags
+    private final ArrayList<Tag> tags = new ArrayList<>();
```

Comment:
Are you sure we won't have any race condition with the clear / fill of an external list?

Reply:
I thought this too. Especially when using multiple rendering threads.

Reply:
Ah. does it mean that MF may read from `MapFile` from multiple threads at once? If you confirm it for me, I'll revert the improvement no 2. , because it lost its benefit in the reduced number of created lists.